lib/utils.pm

```diff
@@ -1120,4 +1121,54 @@ sub prepare_ssh_localhost_key_login {
     }
 }
 
+=head2 script_run_interactive
+For interactive command, input strings or keys according to the prompt message
+in the run time. Pass arrayref which contains the prompt message to be matched
+(regex) and the answers
+=cut
+sub script_run_interactive {
+    my ($cmd, $scan, $timeout) = @_;
+    my $output;
+    my @words;
+    $timeout //= 180;
+
+    if ($cmd) {
+        script_run("script -c \'", 0);
+        script_run($cmd,           0);
+
+        # Write to /dev/null since we want not to leave file there
+        script_run("\' /dev/null |& tee /dev/$serialdev", 0);
+    }
+
+    for my $k (@$scan) {
+        push(@words, $k->{prompt});
+    }
+
+    my $endmark = "Script done.*\/dev\/null";
+    push(@words, $endmark);
+
+  LOOP: {
```

Comment:
I'm not sure if this label helps in anyway, as everyone can see that it's a loop below. So I would either rename it that it explains what the loop does or get rid of it. You could also simply wrap it in the subroutine.

Reply:
Removed label which is not meaningful as you described